libdill.h

```diff
@@ -967,6 +967,29 @@ DILL_EXPORT int dill_ws_response_key(
 #define ws_response_key dill_ws_response_key
 #endif
 
+/******************************************************************************/
+/*  SOCKS5                                                                    */
+/******************************************************************************/
+
+struct dill_socks5_storage {char _[1200];};
+
+DILL_EXPORT int dill_socks5_attach_client(
+    int s, const char *username, const char *password, int64_t deadline);
+DILL_EXPORT int dill_socks5_attach_client_mem(
+    int s, const char *username, const char *password,
+    struct dill_socks5_storage *mem, int64_t deadline);
+DILL_EXPORT int dill_socks5_connect(
+    int s, char *addr, int port, int64_t deadline);
```

Comment:
You can change "addr" parameter to be of "struct ipaddr" type. That would allow the user to use all the address parsing and DNS functionality already present in libdill (see ipaddr_local, ipaddr_remote).